src/node/share_manager.h

```diff
@@ -47,24 +47,37 @@ namespace ccf
       recovery_threshold(recovery_threshold_)
     {
       shares = shares_;
-      ccf::crypto::sharing::Share secret;
       ccf::crypto::sharing::recover_unauthenticated_secret(
         secret, shares, recovery_threshold);
       data = secret.key(KZ_KEY_SIZE);
     }
 
+    LedgerSecretWrappingKey(ccf::crypto::sharing::Share& secret_)
+    {
+      secret = secret_;
+      num_shares = 1;
+      shares.resize(num_shares);
+      shares.emplace_back(secret_);
+      data = secret_.key(KZ_KEY_SIZE);
+    }
+
     LedgerSecretWrappingKey(
       std::vector<SecretSharing::Share>&& shares_, size_t recovery_threshold_) :
       recovery_threshold(recovery_threshold_)
     {
-      auto secret = SecretSharing::combine(shares_, shares_.size());
-      data.resize(secret.size());
-      std::copy_n(secret.begin(), secret.size(), data.begin());
-      OPENSSL_cleanse(secret.data(), secret.size());
+      // TBD (gsinha): How to set ccf::crypto::sharing::Share secret member
+      // variable here?
+      auto combined_secret = SecretSharing::combine(shares_, shares_.size());
+      data.resize(combined_secret.size());
+      std::copy_n(
+        combined_secret.begin(), combined_secret.size(), data.begin());
+      OPENSSL_cleanse(combined_secret.data(), combined_secret.size());
     }
 
     ~LedgerSecretWrappingKey()
     {
+      // TBD (gsinha): Need to cleanse ccf::crypto::sharing::Share secret member
+      // variable here? If so, how?
```

Comment:
Should be taken care of already by https://github.com/microsoft/CCF/blob/c942ccb9b6cf7bb98673470274a17c436276e0aa/src/crypto/sharing.h#L38, and if we remove data, we can drop this.